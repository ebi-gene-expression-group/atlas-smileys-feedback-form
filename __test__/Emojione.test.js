import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Emoji from '../src/Emojione'

Enzyme.configure({ adapter: new Adapter() })

describe(`Emojione`, () => {

  const smileyId = [`disappointed`, `slightly_frowning_face`, `neutral_face`, `grin`, `satisfied`]
  const randoemFace = smileyId[Math.round(Math.random(0,smileyId.length))]
  const props = {
    emoji: randoemFace,
    onLeave: ()=>{},
    onOver: ()=>{},
    onClick: ()=>{},
  }

  test(`should render 2 span components and id equals to emoji id`, () => {
    const wrapper = shallow(<Emoji {...props} />)
    expect(wrapper.find(`span`)).toHaveLength(2)
    expect(wrapper.find(`#${randoemFace}`)).toHaveLength(1)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<Emoji  {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})