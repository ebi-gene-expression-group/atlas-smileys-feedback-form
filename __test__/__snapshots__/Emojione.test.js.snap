// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emojione matches snapshot 1`] = `
<span
  id="grin"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    style={
      Object {
        "fontSize": "40px",
      }
    }
  >
    😁
  </span>
</span>
`;
